### PR TITLE
Speed up New() by factor 30x

### PR DESCRIPTION
**Pull request by mologie**
* Instead of converting polygons for s2 in New(), precompute them in
  datagen and serialize that using s2's built-in methods. This gives
  a significant performance boost, factor 3s to 100ms on my box.

* Use zstd for better compression rate and decompression performance.

* Arbitrary GeoJSON can be converted and loaded on the fly now due to
  a new intermediate Dataset type and constructors to load from
  precomputed data or an arbitrary feature collection struct.

* Type of datasets has changed, API is thus incompatible to v1. Might
  still be drop-in if the type is not used explicitly.

Note that time to build the index is not contained in the old or new number. That can and does happen in the background in AVAS though, and does not block startup.

### CODE REVIEW

File: cmd/datagen/datagen.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+// readGeoJSON parses a GeoJSON file as geojson.FeatureCollection
+func readGeoJSON(path string) (*geojson.FeatureCollection, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("open file: %w", err)
+	}
+	defer func() {
+		_ = f.Close()
+	}()
```
Comment (mboegl) [inline cmd/datagen/datagen.go]: lets condense that into a single line, like in every other `defer`'ed `Close()`

File: regen.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -eu
+
+TAG=v5.1.2
+
+echo "Updating dataset..."
+if ! [ -d natural-earth-vector ]; then
+	git clone https://github.com/nvkelso/natural-earth-vector.git --branch "$TAG"
```
Comment (mboegl) [inline regen.sh]: adding `--depth=1` would suffice and make the download significantly faster

File: cmd/datagen/datagen.go (hunk 1 above)
Comment (mologie) [reply cmd/datagen/datagen.go]: That would leave a warning in GoLand for an ignored error from Close()

Comment (mboegl) [reply cmd/datagen/datagen.go]: I just meant a cosmetic change from the above 3 lines to the single line (like in line 108)

`defer func() { _ = f.Close() }()`

**Comment by mboegl** [comment]
The rest LGTM, but we might wanna take some time in the future (or get an intern^^) to update some references to our fork and update the READMEs.

**Comment by mologie** [comment]
Thanks for the review! Will patch up the bits you mentioned.

RE updating references: `go.mod`'s `replace` will do for now since I intend to use this for AVAS only -- ideally I'd want to upstream all of our changes. Looking forward to the maintainer's response...

**Pull request merged**